Add ur_client_library

fmauch opened the pull request:
We are currently in the process of splitting the client library from the [ur_robot_driver](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver) itself. The client library shall be released first so that users can install it using rosdep instead of having to build their workspace in isolation because the library isn't a catkin package.

fmauch committed:
Add ur_client_library

ivanpauno commented on melodic/distribution.yaml:
```diff
@@ -12140,6 +12140,17 @@ repositories:
+        release: release/melodic/{package}/{version}
+      url: https://github.com/UniversalRobots/Universal_Robots_Client_Library-release.git
+      version: 0.1.0-1
+    source:
```
can you add a doc tab?

fmauch replied:
That would require adding a `rosdoc.yaml` file and `mainpage.dox` file through the release repository as described [here](http://wiki.ros.org/bloom/Tutorials/ReleaseThirdParty#Modifying_the_Release_Repository), right?

I didn't want to clutter the source repository with that, as it is quite ROS specific. On the other hand, we have a `package.xml` file there, already...

ivanpauno replied:
> That would require adding a rosdoc.yaml file and mainpage.dox file through the release repository as described here, right?

I think that's only required when you don't use doxygen (see [here](http://wiki.ros.org/rosdoc)), maybe @cottsay can chime in.

cottsay replied:
Sorry, I have absolutely no experience with the doc generation...

fmauch replied:
As I have already added a doc entry for the [Universal Robots ROS Driver](https://github.com/ros/rosdistro/blob/master/melodic/distribution.yaml#L12174) which created also documentation for contained packages, which don't have a respective `rosdoc.yaml` and `mainpage.dox` such as [ur_calibration](http://docs.ros.org/melodic/api/ur_calibration/html/) it should be fine adding a doc entry without explicitly setting this up.

fmauch replied:
I've added a doc section to this PR.

ivanpauno reviewed (approved):
A few questions before merging, but it lgtm:
- The LICENSE file says Apache, and the package.xml Apache/BSD-2clause/Zlib, is the package multi-licensed? If the other listed licenses are from dependencies, it would be great to clarify that in the package.xml.
- was the release done with bloom?

fmauch commented:
> A few questions before merging, but it lgtm:
> 
> * The LICENSE file says Apache, and the package.xml Apache/BSD-2clause/Zlib, is the package multi-licensed? If the other listed licenses are from dependencies, it would be great to clarify that in the package.xml.

There are upstream components included which are released under the respective licenses. I'll clarify this in more detail.

> * was the release done with bloom?

Yes, however, aquiring the GH token did not work, which is why I made the PR by hand.

ivanpauno commented:
> There are upstream components included which are released under the respective licenses. I'll clarify this in more detail.

Thanks!

> Yes, however, aquiring the GH token did not work, which is why I made the PR by hand.

Sounds good

cottsay reviewed (approved):
This LGTM.

It would be good, however, if your `CMakeLists.txt` installed the `package.xml` you're including in the repository. It looks like maybe the `CMakeLists.txt` is aware if whether it is in a ROS build or not, so maybe you could conditionally install it based on that. If that isn't the case, it might also be possible to patch the `CMakeLists.txt` in the release repository.

fmauch commented:
@ivanpauno I've added a [license section](https://github.com/UniversalRobots/Universal_Robots_Client_Library/tree/boost#license) to the README to clarify all licenses included in this package.

Also, I've added an install statement for the `package.xml` file given that catkin is found.

If you think, that this should be sufficient, I'll release a version 0.1.1 including those changes.

fmauch committed:
added a doc section for the client library, as well.

cottsay reviewed (approved):
> Also, I've added an install statement for the package.xml file given that catkin is found.
> 
> If you think, that this should be sufficient, I'll release a version 0.1.1 including those changes.

Looks great, thanks!

fmauch committed:
Place doc before release and source tags

fmauch committed:
Increased version to 0.1.1

fmauch commented:
Everything should be setup accordingly.

ivanpauno reviewed (approved):
Thanks @fmauch
